testing/helpers/qunitExtensions.js

```diff
@@ -153,6 +153,82 @@
 
 }();
 
+(function setupShadowDomMode() {
+    function getRoot() {
+        return document.querySelector('#qunit-fixture').shadowRoot;
+    }
+
+    function get(selector) {
+        return getRoot().querySelector(selector);
+    }
+
+    function isInShadowDomMode() {
+        return QUnit.urlParams['shadowDom'] && QUnit.urlParams['nojquery'];
+    }
+
+    function addShadowRootTree() {
+        const root = document.querySelector('#qunit-fixture');
+
+        if(!root.shadowRoot) {
+            root.attachShadow({ mode: 'open' });
+        }
+
+        const shadowContainer = document.createElement('div');
+        shadowContainer.part = 'shadow';
+
+        root.shadowRoot.appendChild(shadowContainer);
+    }
+
+    function clearShadowRootTree() {
+        const container = get('div');
+
+        jQuery(container).empty();
```

Comment:
Is removing really not enough?

Reply:
I was related to jQuery and how jQuery clears the elemens inside the selector. However, after some refactoring this line is redundant.